src/system/AdminModule/Controller/AdminInterfaceController.php

```diff
@@ -0,0 +1,469 @@
+        $caller['_zkFunc'] = $masterRequest->attributes->get('_zkFunc');
+        return $this->render("ZikulaAdminModule:AdminInterface:header.html.twig", [
+            'caller' => $caller
+        ]);
+    }
+
+    /**
+     * @Route("/footer")
+     *
+     * Close the admin container
+     *
+     * @return Response symfony response object
+     */
+    public function footerAction()
+    {
+        $masterRequest = $this->get('request_stack')->getMasterRequest();
+        $caller = [];
+        $caller['_zkModule'] = $masterRequest->attributes->get('_zkModule');
+        $caller['info'] = \ModUtil::getInfoFromName($caller['_zkModule']);
+        return $this->render("ZikulaAdminModule:AdminInterface:footer.html.twig", [
+            'caller' => $caller,
+            'symfonyversion' => \Symfony\Component\HttpKernel\Kernel::VERSION,
+            'phpversion' => phpversion()
+        ]);
+    }
+
+    /**
+     * @Route("/breadcrumbs")
+     * @Method("GET")
+     *
+     * Admin breadcrumbs
+     *
+     * @return Response symfony response object
+     */
+    public function breadcrumbsAction()
+    {
+        if (! $this->hasPermission('ZikulaAdminModule::', '::', ACCESS_ADMIN)) {
+            throw new AccessDeniedException();
+        }
+        
+        $masterRequest = $this->get('request_stack')->getMasterRequest();
+        $requested_cid = $masterRequest->attributes->get('acid');
+        $caller = [];
+        $caller['_zkModule'] = $masterRequest->attributes->get('_zkModule');
+        $caller['_zkType'] = $masterRequest->attributes->get('_zkType');
+        $caller['_zkFunc'] = $masterRequest->attributes->get('_zkFunc');
+        $caller['info'] = \ModUtil::getInfoFromName($caller['_zkModule']);
+        
+        if ($caller['_zkModule'] == 'ZikulaAdminModule') {
+            $cid = empty($requested_cid) ? $this->getVar('startcategory') : $requested_cid;
+        } else {
+            $cid = \ModUtil::apiFunc('ZikulaAdminModule', 'admin', 'getmodcategory', array(
+                'mid' => $caller['info']['id']
+            ));
+        }
+        $caller['category'] = \ModUtil::apiFunc('ZikulaAdminModule', 'admin', 'getCategory', array(
+            'cid' => $cid
+        ));
+        
+        return $this->render("ZikulaAdminModule:AdminInterface:breadcrumbs.html.twig", [
+            'caller' => $caller
+        ]);
+    }
+
+    /**
+     * @Route("/developernotices")
+     *
+     * Add developer notices
+     *
+     * @return Response symfony response object
+     */
+    public function developernoticesAction()
+    {
+        if (! $this->hasPermission('ZikulaAdminModule::', '::', ACCESS_ADMIN)) {
+            throw new AccessDeniedException();
+        }
+        
+        $modvars = $this->get('zikula_extensions_module.api.variable')->getAll('ZikulaThemeModule');
+        $data = [];
+        $data['mode'] = $this->get('kernel')->getEnvironment() === 'dev';
+        if ($data['mode'] == 'dev') {
```

Comment:
This makes no sense. In line 121, `$data['mode']` is set to a boolean value and then checked against a string in line 122.